src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3342,15 +3226,8 @@ void c_typecheck_baset::typecheck_expr_pointer_arithmetic(exprt &expr)
 
 void c_typecheck_baset::typecheck_expr_binary_boolean(exprt &expr)
 {
-  if(expr.operands().size()!=2)
-  {
-    error().source_location = expr.source_location();
-    error() << "operator '" << expr.id() << "' expects two operands" << eom;
-    throw 0;
-  }
-
-  implicit_typecast_bool(expr.op0());
-  implicit_typecast_bool(expr.op1());
+  implicit_typecast_bool(to_binary_expr(expr).op0());
+  implicit_typecast_bool(to_binary_expr(expr).op1());
```

Comment:
I think we should do `to_binary_expr` just once.

Reply:
done